### ISSUE TITLE
Model's result is different in Android app

### ISSUE THREAD

**Issue by NusratNB** [sebastian-sz/efficientnet-lite-keras#27]
Hello, thank you for your awesome work.
I trained your model for the custom dataset, and the result is pretty good. After the implementation of the optimization (pruning, quantization ), I checked the result of the model, and it's quite satisfying. However, after implementing it on the Android app, the model shows the wrong result, even on the same test images. 
 
I have used EfficientNetLiteB4 version. I didn't implement any normalization process for input data. Here is the code for Data loader:
`class DataGenerator(keras.utils.Sequence):

    def __init__(self, list_IDs,  batch_size=8, dim=(300, 300), n_channels=3, cl_numbers=11,
                 shuffle=True):
        'Initialization'
        self.dim = dim
        self.batch_size = batch_size        
        self.list_IDs = list_IDs
        self.n_channels = n_channels
        self.shuffle = shuffle
        self.cl_numbers = cl_numbers
        self.on_epoch_end()

    def __len__(self):
        'Denotes the number of batches per epoch'
        return int(np.floor(len(self.list_IDs) / self.batch_size))

    def __getitem__(self, index):
        'Generate one batch of data'
        # Generate indexes of the batch
        indexes = self.indexes[index * self.batch_size:(index + 1) * self.batch_size]

        # Find list of IDs
        list_IDs_temp = [self.list_IDs[k] for k in indexes]

        # Generate data
        X, y = self.__data_generation(list_IDs_temp)

        return X, y

    def on_epoch_end(self):
        'Updates indexes after each epoch'
        self.indexes = np.arange(len(self.list_IDs))
        if self.shuffle == True:
            np.random.shuffle(self.indexes)

    def __data_generation(self, list_IDs_temp):
        'Generates data containing batch_size samples'

        X = np.zeros((self.batch_size, *self.dim, self.n_channels))
        y = np.zeros((self.batch_size, self.cl_numbers))
        

        # Generate data
        for i in range(len(list_IDs_temp)):
            ID = list_IDs_temp[i]
            id_split = ID.split(" ")
            cl_ind = id_split[1]
            pth_to_img = id_split[0]
            specto_img = cv2.imread(pth_to_img)
            res_specto_img = cv2.resize(specto_img, (300, 300), interpolation=cv2.INTER_CUBIC)
            
            X[i,] = res_specto_img
            

            y[i,int(cl_ind)] = 1
        return X, y`

Model initialization:

`model = EfficientNetLiteB4(weights=None, input_shape=(300, 300, 3))`

`def EfficientNetLiteB4(
    include_top=True,
    weights="imagenet",
    input_tensor=None,
    input_shape=None,
    pooling=None,
    classes=11,
    classifier_activation="softmax",
    **kwargs,
):
    """Create Efficient Net Lite B4 variant."""
    return EfficientNetLite(
        1.4,
        1.8,
        300,
        0.3,
        model_name="efficientnetlite4",
        include_top=include_top,
        weights=weights,
        input_tensor=input_tensor,
        input_shape=input_shape,
        pooling=pooling,
        classes=classes,
        classifier_activation=classifier_activation,
        **kwargs,
    )
`

Settings on [metadata_writer](https://github.com/tensorflow/examples/blob/master/lite/examples/image_classification/metadata/metadata_writer_for_image_classifier.py):

`_MODEL_INFO = {
    "actual_model_jpeg_1130.tflite":
        ModelSpecificInfo(
            name="EfficientNetLite image classifier",
            version="v4",
            image_width=300,
            image_height=300,
            image_min=0.0,
            image_max=255.0,
            mean=[0.0],
            std=[1.0],
            num_classes=11,
            author="Keepin")
}
`

**Comment by NusratNB** [comment]
I'm assuming that the pixel values of the input data on the Android app are changing (normalizing) at some point. As far as I know, there is no default normalization at your repo.

**Comment by sebastian-sz** [comment]
@NusratNB 
Hello, and thanks for the kind words!

Quick question about the Android App:
Are you passing RGB or BGR images?

In your training code I can see you are loading, using OpenCV:
```python
specto_img = cv2.imread(pth_to_img)
res_specto_img = cv2.resize(specto_img, (300, 300), interpolation=cv2.INTER_CUBIC)
```

By default, OpenCV loads images in BGR format, **not RGB** -  if you do not believe me, you can use matplotlib for preview:
```python
specto_img = cv2.imread(pth_to_img)
res_specto_img = cv2.resize(specto_img, (300, 300), interpolation=cv2.INTER_CUBIC)
plt.imshow(res_specto_img)
plt.show()
```
OpenCV will not show you this, as it's automatically tranposing the channels during functions like `cv2.imshow` or `cv2.imwrite`.

If in your Android App you are using RGB images (which is a standard outside OpenCV) the difference might come from this.

**Comment by sebastian-sz** [comment]
@NusratNB 
To load RGB images, during training / eval, you can add transpose to your code:

```python
specto_img = cv2.imread(pth_to_img)
res_specto_img = cv2.resize(specto_img, (300, 300), interpolation=cv2.INTER_CUBIC)
res_specto_img = cv2.cvtColor(res_specto_img, cv2.COLOR_BGR2RGB)
```

**Comment by NusratNB** [comment]
@sebastian-sz 
Thank you for your reply. 
Your answer completely makes sense. I guess I have to retrain the model. After, receiving the result I will let you know.
Thank you again.

**Comment by sebastian-sz** [comment]
@NusratNB 
No problem, let me know if this helps.

**Comment by NusratNB** [comment]
> @NusratNB To load RGB images, during training / eval, you can add transpose to your code:
> 
> ```python
> specto_img = cv2.imread(pth_to_img)
> res_specto_img = cv2.resize(specto_img, (300, 300), interpolation=cv2.INTER_CUBIC)
> res_specto_img = cv2.cvtColor(res_specto_img, cv2.COLOR_BGR2RGB)
> ```

@sebastian-sz this solution solved the error. Thank you again:)

**Comment by sebastian-sz** [comment]
@NusratNB I'm happy that it helped!
